main.ts

```diff
@@ -122,6 +122,23 @@ class Translator {
     return true;
   }
 
+  visitNamedParameter(paramDecl: ts.ParameterDeclaration) {
+    this.visitEachIfPresent(paramDecl.decorators);
+    if (paramDecl.type) {
+      // TODO(martinprobst): These are currently silently ignored.
+      // this.reportError(paramDecl.type, 'types on named parameters are unsupported');
+    }
+    this.visit(paramDecl.name);
+    if (paramDecl.initializer) {
```

Comment:
I think you're saying there's another validation we could add? I didn't quite follow. Maybe you're saying there's a bug in this code?